src/Sylius/Bundle/ShopBundle/Twig/Component/Common/ProductCardComponent.php

```diff
@@ -41,10 +41,13 @@ public function __construct(
     #[ExposeInTemplate(name: 'variant')]
     public function getProductVariant(): ?ProductVariantInterface
     {
-        /** @var ProductVariantInterface|null $variant */
         $variant = $this->productVariantResolver->getVariant($this->product);
 
-        if (null === $variant) {
+        if (!$variant instanceof ProductVariantInterface) {
+            $variant = $this->product->getVariants()->first();
+        }
```

Comment:
Is this check needed here? `ProductVariantResolver` should resolve variant for a product 🤔

Reply:
it is becouse of this scenario: https://github.com/Sylius/Sylius/blob/bootstrap-shop/features/shop/product/viewing_products/viewing_product_with_all_variants_disabled.feature#L20-L23